front/app/components/UI/QuillEditor/altTextToImagesModule.tsx

```diff
@@ -0,0 +1,149 @@
+
+const Module = Quill.import('core/module');
+const BlockEmbed = Quill.import('blots/block/embed');
+
+export const attributes = ['alt', 'width', 'height', 'style'];
+
+// Create a custom ImageBlot that allows us to add alt text to the image
+export class ImageBlot extends BlockEmbed {
+  static blotName = 'image';
+  // Instead of using the default img tag, we are using a div where we can add an input field for the alt text
+  static tagName = ['div'];
+
+  static create(value: { alt: string; src: string } | string) {
+    const node = super.create();
+    // When we are creating the image blot, we need to create and add an img tag to it with the necessary attributes
+    const img = window.document.createElement('img');
+    if (typeof value === 'string') {
+      img.setAttribute('src', value);
+      img.setAttribute('alt', '');
+    } else {
+      img.setAttribute('alt', value.alt);
+      img.setAttribute('src', value.src);
+    }
+    // We are appending the img tag to the div that is the blot
+    node.appendChild(img);
+    // We are setting a custom class on the div that is the blot so that we can reference it later on
+    node.setAttribute('class', 'ql-alt-text-input-container');
+
+    return node;
+  }
+
+  constructor(
+    domNode: HTMLDivElement & { onSelect: () => void; onDeselect: () => void }
+  ) {
+    super(domNode);
+    const img = domNode.querySelector('img');
+    // We are creating an input field for the alt text and setting the necessary attributes
+    const altInput = window.document.createElement('input');
+    altInput.setAttribute('type', 'text');
+    altInput.setAttribute('class', 'ql-alt-text-input');
+    altInput.setAttribute(
+      'style',
+      'display: block; width:100%; border: 1px solid #ccc; border-radius: 3px; padding: 5px; margin-bottom:4px;'
+    );
+    altInput.setAttribute('value', img?.getAttribute('alt') || '');
+    // Handle change of input field value
+    const handleChange = (e: Event) => {
+      const target = e.target as HTMLTextAreaElement;
+      img?.setAttribute('alt', target.value);
+    };
+
+    // We are adding the input field to the blot
+    domNode.prepend(altInput);
+    // When the blot is selected, we are adding the event listener to the input field
+    domNode.onSelect = () => {
+      altInput.addEventListener('input', handleChange);
+      altInput.focus();
+    };
+    // When the blot is deselected, we are removing the event listener from the input field
+    domNode.onDeselect = () => {
+      altInput.removeEventListener('input', handleChange);
+    };
+  }
+  // Value method returns the value of src and the alt attribute
+  static value(domNode: HTMLElement) {
+    const img = domNode.querySelector('img');
+    if (!img) return false;
+    return {
+      alt: img.getAttribute('alt'),
+      src: img.getAttribute('src'),
+    };
+  }
+
+  static formats(domNode: HTMLDivElement) {
+    // Register unregistered embed formats comming from attributes
```

Comment:
Embed formats such as?

Reply:
Improved this comment to hopefully make it more clear 👍